File: app/src/main/java/com/example/messenger/MainViewModel.java
package com.example.messenger;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainViewModel extends ViewModel {

    private final MutableLiveData<String> error = new MutableLiveData<>();
    private final MutableLiveData<FirebaseUser> user = new MutableLiveData<>();
    private final FirebaseAuth auth;

    public MainViewModel() {
        auth = FirebaseAuth.getInstance();
        auth.addAuthStateListener(firebaseAuth -> {
            if(auth.getCurrentUser() != null) {
                user.setValue(auth.getCurrentUser());
            }
        });
    }

    public LiveData<FirebaseUser> getUser() {
        return user;
    }

    public LiveData<String> getError() {
        return error;
    }

    public void login(String email, String password) {
        auth.signInWithEmailAndPassword(email, password)
                .addOnFailureListener(e -> error.setValue(e.getMessage()));
    }
}

File: app/src/main/java/com/example/messenger/MainViewModel.kt
package com.example.messenger

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser

class MainViewModel : ViewModel() {
    private val error: MutableLiveData<String> = MutableLiveData()
    private val user: MutableLiveData<FirebaseUser> = MutableLiveData()
    private val auth = FirebaseAuth.getInstance()

    init {
        auth.addAuthStateListener {
            auth.currentUser?.let {
                user.value = auth.currentUser
            }
        }
    }

    val getUser: LiveData<FirebaseUser> get() = user
    val getError: LiveData<String> get() = error

    fun login(email: String, password: String) {
        auth.signInWithEmailAndPassword(email, password)
                .addOnFailureListener { error.value = it.message }
    }
}